burst/src/K8sApi/K8sHPAMetricsService.cs

```diff
@@ -61,11 +62,25 @@ public K8sHPAMetrics GetK8SHPAMetrics(string ns, string deployment)
                 {
                     if (hpa.Namespace().Equals(ns) && hpa.Name().Equals(deployment))
                     {
-                        // Get the Target CPU load
-                        hpaMetrics.TargetCPULoad = GetTargetCpuLoad(hpa);
+                        try
+                        {
+                            // Get the Target CPU load
+                            hpaMetrics.MaxLoad = GetMaxLoad(hpa);
+
+                            // Get the current CPU load
+                            hpaMetrics.CurrentLoad = GetCurrentLoad(hpa);
+                            hpaMetrics.TargetLoad = (int?)Math.Floor(hpaMetrics.MaxLoad.GetValueOrDefault() * TargetPercent);
+                        }
+                        catch (Exception ex)
+                        {
+                            logger.LogWarning(ex.Message);
+                        }
```

Comment:
If an error is caught for any of those value assignments then we need to set all of the values to the invalid value (right now you're using -1).  That way in the final condition before returning we can test and see that we should return a 204.  Also, create a constant for invalid value, maybe something like, `private const int INVALID_METRIC_VALUE` then you can use that for assignment and test and we have a single place to change it if we need to change it.

Reply:
`hpaMetrics` members will remain null without any value assignmentsm since it's initialized at line 51.

Do we still need to set MaxLoad, CurrentLoad and TargetLoad to an `INVALID_METRIC_VALUE` to check in BurstMetricsService?

Reply:
Depends on how you test. If the first and second assignment succeed but the third fails then you have a two valid values that will be returned. If, especially under maintenance, the check is not correct then you could end up sending back partially correct data.